### ISSUE TITLE
1.19 update and future?

### ISSUE THREAD

**Issue by PweSol** [Hephaestus-Dev/Atmosfera#36]
Hello, can we get an update to 1.19 please?

Are there any plans to continue adding music from MC Dungeons? There have been a number of updates for MCD but nothing else was added to this mod. It it just lack of interest in updating this mod or is it something like time constraints? Because I would love to see more updates to this mod.

**Comment by Haven-King** [comment]
1.18.1 threw a lot of wrenches into the systems Atmosfera uses and updating it would be a huge burden. I don't have the time to commit to this project anymore but would be open to somebody else with the skills to do so taking over.

**Comment by EGOIST1372** [comment]
@Haven-King I did ask **Fourmisain** to help out and he seems can help and he is a great dev with a good spirit. I hope you all can get along and continue this journey.
Thanks both of you guys <3 <3

**Comment by Haven-King** [comment]
@Fourmisain I would be happy to have you take over as maintainer of this project! I am out of town for the next few days and unable to test your fork, but if you open a PR just so we keep the changes all together, I will try to review it this weekend and add you to both the GitHub and CurseForge projects.

**Comment by Zetabite** [comment]
Regarding the future of Atmosfera, i will also try to do my bit, as I like the mod too, and would like to see it more commonly used

**Comment by EGOIST1372** [comment]
I guess illumination was the culprit of that crash but IDK maybe konkret update helped too

**Comment by EGOIST1372** [comment]
IDK why I get so many memory leak. I even installed [memory leak fix mod](https://www.curseforge.com/minecraft/mc-mods/memoryleakfix) to combat any possible memory leak. but still there are so many logs like this:
[21:35:07] [Render thread/WARN]: Reclaimed 32720 bytes at address 2233329976128 that were leaked from an unknown location (logging is disabled)
I get so many sound errors like:
[21:35:23] [Render thread/WARN]: Unable to play empty soundEvent: minecraft:entity.salmon.ambient
[21:54:28] [Render thread/WARN]: Unable to play empty soundEvent: minecraft:entity.cod.ambient
[21:50:10] [Render thread/WARN]: Unable to play empty soundEvent: minecraft:entity.tropical_fish.ambient
[21:50:11] [Render thread/WARN]: Unable to play empty soundEvent: minecraft:entity.puffer_fish.ambient
also getting this error :
[21:55:20] [Server thread/WARN]: Can't keep up! Is the server overloaded? Running 8167ms or 163 ticks behind
[21:54:57] [Server thread/WARN]: Can't keep up! Is the server overloaded? Running 2242ms or 44 ticks behind

full log file:
[latest.log](https://github.com/Haven-King/Atmosfera/files/8987150/latest.log)

**Comment by Fourmisain** [comment]
I still doubt your memory leak is caused by Atmosfera, I have done some testing on this and didn't run into any issues, though I also cannot fully dismiss it - but not for the reasons you might be thinking. Sounds not ending and stuff like that will **not** cause a memory leak.

Try to get rid of Farsight first and retest, that's still the most likely candidate.

About those warnings:

> [21:35:07] [Render thread/WARN]: Reclaimed 32720 bytes at address 2233329976128 that were leaked from an unknown location (logging is disabled)

That's Sodium trying it's best to fix the mess.

> Unable to play empty soundEvent: minecraft:entity.salmon.ambient

Those are completely normal, it's an [**ancient** bug](https://bugs.mojang.com/browse/MC-97521).

> [21:55:20] [Server thread/WARN]: Can't keep up! Is the server overloaded? Running 8167ms or 163 ticks behind
[21:54:57] [Server thread/WARN]: Can't keep up! Is the server overloaded? Running 2242ms or 44 ticks behind

This just means the server is temporarily running slow. If your memory is nearly full and your game is chugging, this is to be expected.
There are some cases where these warnings are completely normal, e.g. after booting up a dedicated server or right here in your log:

> [21:48:44] [Server thread/WARN]: Can't keep up! Is the server overloaded? Running 10014ms or 200 ticks behind
[21:48:44] [Render thread/INFO]: [CHAT] **The nearest minecraft:desert is at [2985, 81, 3924] (4802 blocks away)**
[21:48:44] [Server thread/INFO]: [EGOIST23: Set own game mode to Creative Mode]
[21:48:44] [Render thread/INFO]: [CHAT] Set own game mode to Creative Mode
[21:48:54] [Render thread/INFO]: [CHAT] The nearest minecraft:desert is at [2985, 81, 3924] (4802 blocks away)
[21:49:00] [Server thread/INFO]: [EGOIST23: Teleported EGOIST23 to 2985.500000, 81.000000, 3924.500000]
[21:49:00] [Render thread/INFO]: [CHAT] **Teleported EGOIST23 to 2985.500000, 81.000000, 3924.500000**
[21:49:03] [Server thread/WARN]: Can't keep up! Is the server overloaded? Running 3086ms or 61 ticks behind

You used `/locate`, which is an expensive command (because it needs to search surroundings for your biome) and followed it up with `/teleport`, which loads/generates a bunch of new chunks, so it's normal for the server to run behind.

Again, please try disabling mods one-by-one and do long play sessions to get an idea if _one particular_ mod is causing the issue or not.
Logs don't help narrow down the cause, because the memory leaks are invisible.

**Comment by EGOIST1372** [comment]
> This just means the server is temporarily running slow. If your memory is nearly full and your game is chugging, this is to be expected

I have 32GB ram and it's not even half used when I launch minecraft and I already dedicated 6GB to it. it can't be nearly full tho
but even after I teleported somewhere and let it settle and loads chunks around still has problem and is going down to even 2 fps ;-;

BTW I removed farsight and replaced phosphor with starlight and it seems fps is now 150+ even tho the chunk I'm in is not loadedd yet
![image](https://user-images.githubusercontent.com/86555131/175830091-767fd90b-f403-44fe-afb7-04cc297453de.png)

it might be phosphor or farsight. but now it's much better
thanks so much <3

**Comment by Fourmisain** [comment]
> I have 32GB ram and it's not even half used when I launch minecraft and I already dedicated 6GB to it. it can't be nearly full tho

By "memory nearly full" I mean the game's heap memory, you can see it if you press F3 on the top right. You'll see the numbers quickly rise and fall down again. When it's constantly at high values, you know very little free space is left and that's where the game starts chugging (and crashing).

> even tho the chunk I'm in is not loadedd yet

I had that very chunk loading issue too in All of Fabric 5, not sure what that's caused by.

> it might be phosphor or farsight. but now it's much better

It's very unlikely it's Phosphor.
I saw Farsight had some reported issues about memory usage and potential leaks, though nothing came of it.

If you still want high view distance, you can give [Bobby](https://www.curseforge.com/minecraft/mc-mods/bobby) a shot.

**Comment by EGOIST1372** [comment]
> I had that very chunk loading issue too in All of Fabric 5, not sure what that's caused by.

So it's not just me.
thanks for the info and help you been provided for me. I really appreciated <3

**Comment by EGOIST1372** [comment]
IDK if this kinda "add support" need to be done from that side of development or this side

**Comment by EGOIST1372** [comment]
@Fourmisain btw have you seen [charmonium](https://www.curseforge.com/minecraft/mc-mods/charmonium) mod? it has really good performance and sounds pick up really fast(biome and elevation detection is really fast)
can you check the mod to see can we improve atmosfera with that? if you have time
thanks <3

**Comment by EGOIST1372** [comment]
@Haven-King @Fourmisain any news on the state of this mod and future plans?
thanks in advance <3

**Comment by Fourmisain** [comment]
Haven't come around porting to 1.19.3 yet, not sure when I'll be able to. This might take a little more effort since this is where Mojang started putting major technical changes into minor patches, which has it's pros and cons - though I feel like 1.19.2 is still perfectly viable to use in the meantime.
Apart from that, I / we are waiting on Haven-King.

**Comment by Nebby-56** [comment]
Any chance for a 1.20+ update?

**Comment by tajemniktv** [comment]
Hey, any info or willingness for 1.21(.1) update? Pity that the main dev *died* and didn't add people willing to update it as maintainers, as it kinda made Atmosfera dead... And it was such a great mod. It still is. Sadly - without updates. 

Anyways, 1.20 version works (?) on 1.21, except the mod menu config is borked, not sure how it plays out in game tho

**Comment by Fourmisain** [comment]
The mod was surprisingly unaffected by the massive NBT changes in 1.20.5 and all that broke in 1.21.1 was literally just
`new Identifier()` -> `Identifier.of()`

Here's the port: #40
Release found inside as always!

**Comment by tajemniktv** [comment]
Bro, did you just update a mod in less than an hour of some random dude asking online? 😭 

I'm so gonna test it rn lmao

**Comment by tajemniktv** [comment]
Okay, so now that this one awesome person updated Atmosfera up to 1.21, how does it look like when used with Charmonium and/or MAmbience. I used it as a replacement, and sometimes along with Atmosfera, but not quite sure how well they go together. I do know there is stated conflict in mod json, but it'ws only conflicting, not breaking, so not sure if I should/can use it together... 

(Also still confused as to how you just did that update so quickly lmaooo, like that's crazyXD)

**Comment by EGOIST1372** [comment]
Fourmisain is a great dev that I asked to help to keep this mod alive and
he is just awesome at it
Many thanks to him
<3 <3 <3

**Comment by Fourmisain** [comment]
Haven-King just added me as a maintainer and gave me push access on this repo!

I'll start merging the PRs and publishing the releases to CurseForge.

@Haven-King A bit of revenue share would be nice to have too if you haven't!
Also, should I create a page on Modrinth too? People will wanna have it on there too.

**Comment by Haven-King** [comment]
@Fourmisain I've already assigned 100% of future revenue to you on CurseForge 🙂 You are welcome to create a Modinth page as well if you wish.

**Comment by Fourmisain** [comment]
Thanks a ton!
Will create a Modrinth page soon too!

**Comment by Fourmisain** [comment]
[Modrinth page](https://modrinth.com/mod/atmosfera) is up and is already approved!
